python/ray/data/aggregate.py

```diff
@@ -103,6 +105,60 @@ def __init__(self, on: AggregateOnT = None):
             name=(f"mean({str(on)})"))
 
 
+class Std(AggregateFn):
+    """Defines standard deviation aggregation."""
+
+    # Uses Welford's online method for an accumulator-style computation of the
+    # standard deviation. This method was chosen due to it's numerical
+    # stability, and it being computable in a single pass. See
+    # https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm
+
+    def __init__(self, on: AggregateOnT = None, ddof: int = 1):
+        on_fn = _to_on_fn(on)
+
+        def accumulate(a: List[float], r: float):
+            # Accumulates the current count, the current mean, and the sum of
+            # squared differences from the current mean (M2).
+            M2, mean, count = a
+            # Select the data on which we want to calculate the mean.
+            val = on_fn(r)
+
+            count += 1
+            delta = val - mean
+            mean += delta / count
+            delta2 = val - mean
+            M2 += delta * delta2
+            return [M2, mean, count]
+
+        def merge(a: List[float], b: List[float]):
```

Comment:
Add a reference to https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm ?